src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs

```diff
@@ -825,6 +825,30 @@ internal static void ValidateSpecifiedAuthentication(PSCredential credential, st
 
         #endregion
 
+        /// <summary>
+        /// Parse a hostname used with SSH Transport to get embedded
+        /// username and/or port.
+        /// </summary>
+        /// <param name="hostname">host name to parse</param>
+        /// <param name="host">resolved target host</param>
+        /// <param name="userName">resolved target user name</param>
+        /// <param name="port">resolved target port</param>
+        protected void ParseSshHostName(string hostname, out string host, out string userName, out int port)
+        {
+            var uri = new System.Uri("ssh://" + hostname);
```

Comment:
I'd like for this to support absolute URIs `ssh://user@host:822`. I recommend trying to parse Hostname as a Uri, check for a schema, if one exists, use that uri. If one does not, do as you have here.

```csharp
Uri uri;
if (!Uri.TryCreate(hostname, UriKind.Absolute, out uri))
{
    uri = new Uri("ssh://" + hostname);
}
```

Reply:
There is an existing `-ConnectionUri` I would love to use to support this.  However, that parameter has expectations that WSMan is used (like support `-Credential` which we currently aren't able to support with SSH due to how SSH works and prompts for credentials).  It may be worthwhile to have a separate PR to update `-ConnectionUri` to support an absolute uri for SSH and perhaps throw an unsupported exception if any WSMan only parameters are used.  I don't think it's necessary for this PR as the intent to have `hostname` syntax align with common `ssh.exe` usage.

Reply:
So why we fix `hostname` if `ConnectionUri` is more suitable for the scenario?

Reply:
`-ConnectionUri` requires URI syntax which means including a `scheme` like "ssh://".  This goes against the intent of providing consistent syntax with ssh.exe

Reply:
I don't understand why we can not do:
```powershell
invoke-command -ConnectionUri "https://steve@mycomputer"
invoke-command -ConnectionUri "ssh://steve@mycomputer"
```
https://www.iana.org/assignments/uri-schemes/uri-schemes.xhtml
https://tools.ietf.org/id/draft-salowey-secsh-uri-00.html

Reply:
@iSazonov we can certainly support that, but I think that's a separate PR.

Reply:
I'd prefer to keep `-hostname` simple and enhance `-ConnectionUri` - seems it is better UX.